sympy/solvers/ode.py

```diff
@@ -4003,6 +4011,37 @@ def _frobenius(n, m, p0, q0, p, q, x0, x, c, check=None):
 
     return frobdict
 
+def _check_substitution_type_match(eq, func):
+    #Matches any differential equation that order_reducing_substitution can solve.
+    #For this to work equation should min order of derivative to be 1, i.e., f(x) should not be present
+    x = func.args[0]
+    f = func.func
+    order_to_subs = 0
+    D = Dummy()
+    for nsubs_try in range(1,ode_order(eq,f(x))):
+        if eq.subs(f(x).diff(x, nsubs_try), D).has(f(x)):
+            break
+        else:
+            order_to_subs += 1
+    if order_to_subs > 0:
+        return {'var':order_to_subs, 'solutions':True}
+    else:
+        return {'var':order_to_subs, 'solutions':False}
+
+# Use repeated substitution until we do not have a function independent of derivative
+def ode_order_reducing_substitution(eq, func, order, match):
+    r'''
+    Substitutes lowest order derivate in equation to function with order of derivative as 0.Eg f^(n)(x) = g(x), where n is the least order derivate.
```

Comment:
please wrap lines to no more than 72 characters

math should be wrapped in single or double backticks like
```
`this` or ``this``
```

Reply:
Please check for further amendments.